src/v/resource_mgmt/memory_groups.h

```diff
@@ -31,6 +31,12 @@ struct compaction_memory_reservation {
     double max_limit_pct{100.0};
 };
 
+struct partitions_memory_reservation {
+    size_t max_limit_pct = 0;
```

Comment:
Why is this set to 0, then overritten after construction in memory_groups()? Can't we just set at construction as a designated initializer?

Reply:
No particular reason. We can change it.